spec/controllers/course/groups_controller_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe Course::GroupsController, type: :controller do
+  let(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    let(:admin) { create(:administrator) }
+    let(:course) { create(:course, creator: admin) }
+    let(:group) { create(:course_group, course: course) }
+    before { sign_in(admin) }
+
+    describe '#update' do
+      context 'when user is present' do
+        let(:user_to_add) { create(:user) }
+        let(:group_attributes) do
+          id_not_taken = Course::GroupUser.maximum(:id).next.to_s
```

Comment:
If `accepts_nested_attributes` find the id does not exists in db, it will create a new record for me, otherwise it will update existing record.